src/plugins/discover/public/application/main/services/discover_app_state_container.ts

```diff
@@ -69,6 +69,12 @@ export interface DiscoverAppStateContainer extends ReduxLikeStateContainer<Disco
    * @param replace
    */
   update: (newPartial: DiscoverAppState, replace?: boolean) => void;
+  /**
+   * Updates the state in URL first, if replace is true, a history.replace is performed instead of history.push
+   * @param newPartial
+   * @param replace
+   */
+  updateUrlState: (newPartial: DiscoverAppState, replace?: boolean) => void;
```

Comment:
Switching the data view should update the URL state first, but push to history, not replace it. This new function provides this functionality.